inspirehep/modules/search/walkers/elasticsearch.py

```diff
@@ -117,12 +119,17 @@ def query(keyword):
                 node.value = node.value[len('recid:'):]
 
             fields = self.get_fields_for_keyword(keyword, mode='a')
-            if fields == ['authors.full_name', 'authors.alternative_names']:
-                return Q('bool', should=[
-                    Q("match", authors__name_variations=str(node.value)),
-                    Q("match", authors__full_name=str(node.value)),
-                    Q("match", authors__inspire_bai=str(node.value))
-                ])
+            if fields == current_app.config['SEARCH_ELASTIC_KEYWORD_MAPPING']['author']:
+                return Q(
+                    'bool',
+                    must=Q('bool', should=[
+                        Q("match", authors__name_variations=str(node.value)),
+                        Q("term", authors__inspire_bai=str(node.value))
```

Comment:
so, should we make `ids` not analyzed in order to be able to match them with `term` queries? 

Also we don't need to copy to `author` and `exactauthor` anymore?